docs/keywords.md

```diff
@@ -0,0 +1,56 @@
+  # NL-Augmenter Keywords
+
+Every transformation should have keywords that identify the type of transformation that is being written. The following list can be used as reference.
+- the type of linguistic change that the transformation is attempting eg. morphological, lexical, syntactic, word-order, discourse, noise etc.
+- the type of algorithm that is written eg. rule-based, model-based, api-based, external-knowledge-based, transformer-based, parser-based, etc.
+- the naturalness of the generation eg. natural-sounding, natural-looking, unnatural, etc.
+- the potential accuracy & precision of the generation eg. accurate, near-accurate, high-precision, high-coverage, etc.
+- the text-specific modality being covered (if applicable) eg. visual, aural, written, etc.
+- the skills required to differentiate between the input and the output of the transformation (if applicable) eg. causal-reasoning, visual-reasoning, social-reasoning, humor, figurative-language, grammar, world-knowledge, etc. 
+- other transformation specific keywords
+
+Some of the categories can be subjective and they are meant to be used as best estimated. It is okay to include multiple of each type. Example, many transformer based generators might fit into most of the linguistic categories.
+
+
+Keyword | Description
+------- | -----------
+**Type of Linguistic Change** |
+  `morphological` | character level or inflectional/morphological changes
+  `lexical` | the transformation substitutes different words or phrases
+  `syntactic` | the transformation changes the syntax of the input (eg. active to passive)
+  `word-order` | if the order of the words or phrases is changed (eg. topicalisation, changing the order of semantic roles)
+  `discourse` | operating on or generating units of language longer than a sentence (eg. paragraph, multiple utterances in a dialog setting, etc)
+  `noise` | The generation adds random noise which would perturb the examples in a non-traditional manner (eg. removing punctuations, changing case, repeating words/characters)
+ 
+**Type of algorithm** |
+  `rule-based` | if the implementation uses a heuristic or rules based approach
+  `model-based` | if the implementation uses a machine learning model
+  `api-based` | if the implementation uses an external api or tool
+  `external-knowledge-based` | if the implementation uses a corpora, a separate knowledge base, etc.
+  `transformer-based` | if the implementation uses a transformer based model (eg. BERT, T5)
+  `parser-based` | if the implementation uses any syntactic (dependency, constituency, etc) or shallow semantic parser or other semantic parsers (eg. semantic role labelling)
```

Comment:
Are we expecting people to pick multiple types here?
e.g. If I'm making a call to an existing model service online, would it count as both `model-based` and `api-based`? And would `external-knowledge` count as a specific kind of `rule-base`, and `parser/transformer` part of `model`?

If so, maybe we can explicitly say "feel free to pick all that applies"?

Reply:
Yes, multiple types should be possible. In your example, we should consider both. 

I had mentioned about multiple types in line 12, but I'll make it more explicit. I have changed it.